Spigot-API-Patches/0039-LootTable-API.patch

```diff
@@ -0,0 +1,271 @@
+From 6a5c581130263c8492fff59626da341bb8e68ffd Mon Sep 17 00:00:00 2001
+From: Aikar <aikar@aikar.co>
+Date: Sun, 1 May 2016 15:19:49 -0400
+Subject: [PATCH] LootTable API
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/loottable/Lootable.java b/src/main/java/com/destroystokyo/paper/loottable/Lootable.java
+new file mode 100644
+index 0000000..f6f61dc
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/loottable/Lootable.java
+@@ -0,0 +1,36 @@
++package com.destroystokyo.paper.loottable;
++
++public interface Lootable {
++
++    /**
++     * Gets the name of the Loot Table to be used in the World Folder
++     * @return The name, or null if no loot table exists
++     */
++    String getLootTableName();
++
++    /**
++     * Sets the name of the Loot Table to be used in the World Folder
++     * Will use a random seed
++     *
++     * @param name name in either foo or minecraft:foo format
++     * @return The previous Loot Table before the change
++     */
++    default String setLootTable(String name) {
++        return setLootTable(name, null);
++    }
++
++    /**
++     * Sets the name of the Loot Table to be used in the World Folder
++     * Uses supplied Seed
++     *
++     * @param name name in either foo or minecraft:foo format
++     * @param seed seed for the loot table. If null, seed will be random
++     * @return The previous Loot Table before the change
++     */
++    String setLootTable(String name, Long seed);
```

Comment:
I  don't like using wrapper types for its null functionality.
Since we already do `seed != null ? seed : 0L` internally (as will most impls), can't we move the default method to the implementation and do the '0L' there?